python/ray/experimental/workflow/workflow_access.py

```diff
@@ -106,44 +116,77 @@ class WorkflowManagementActor:
 
     def __init__(self, store: "storage.Storage"):
         self._store = store
-        self._workflow_outputs: Dict[str, ray.ObjectRef] = {}
+        self._workflow_outputs: Dict[str, LatestWorkflowOutput] = {}
+        # Cache step output.
+        self._step_output_cache: Dict[str, LatestWorkflowOutput] = {}
         self._actor_initialized: Dict[str, ray.ObjectRef] = {}
         self._step_status: Dict[str, Dict[str, common.WorkflowStatus]] = {}
 
     def get_storage_url(self) -> str:
         """Get hte storage URL."""
         return self._store.storage_url
 
-    def run_or_resume(self, workflow_id: str) -> ray.ObjectRef:
-        """Run or resume a workflow.
+    def get_cached_step(self, workflow_id: str,
+                        step_id: "StepID") -> ray.ObjectRef:
+        """Get the cached result of a step.
 
         Args:
             workflow_id: The ID of the workflow.
+            step_id: The ID of the step.
 
         Returns:
             An object reference that can be used to retrieve the
-            workflow result.
+            step result. If it does not exist, return None
         """
-        if workflow_id in self._workflow_outputs:
+        try:
+            return self._step_output_cache[workflow_id, step_id].output
+        except Exception:
+            return None
+
+    def run_or_resume(self, workflow_id: str, ignore_existing: bool = False
+                      ) -> "WorkflowExecutionResult":
+        """Run or resume a workflow.
+
+        Args:
+            workflow_id: The ID of the workflow.
+            ignore_existing: Ignore we already have an existing output.
```

Comment:
When set false, raise an exception if there has already been a workflow with this id running.